widget/embedded/src/store/slices/data.ts

```diff
@@ -235,11 +278,17 @@ export const createDataSlice: StateCreator<
       // Sort
       blockchains.sort((a, b) => a.sort - b.sort);
 
+      const blockchainNameMap: BlockchainNameMap = {};
```

Comment:
How you think about the following approach:
Change `_blockchains: BlockchainMeta[];` to ` _blockchains: Map<string, BlockchainMeta>;` in our state.
Because `Map` is iterable it can easily convert to array which expected in `blockchains` method/selector. 
Then you only need to use something like this in `blockchains` selector to not break anything:
```js
const blockchainsFromState = get()._blockchains;
const output = Array.from(blockchainsFromState.values())
````
P.s feel free to use more appropriate names for variables.

Reply:
I think it's a good idea

Reply:
Fixed